datacommons/populations.py

```diff
@@ -12,180 +12,97 @@
-      populations belong to.
-    new_col_name: New column name.
-    population_type: Population type like "Person".
-    max_rows: The maximum number of rows returned by the query results.
-    **kwargs: keyword properties to define the population.
-
-  Raises:
-    ValueError: when input argument is not valid.
-  """
-  if seed_col_name not in self._dataframe:
-    raise ValueError('{} is not a valid seed column.'.format(seed_col_name))
-  if new_col_name in self._dataframe:
-    raise ValueError('{} is already a column name.'.format(new_col_name))
+def get_populations(dcids, population_type, pv_map, reload=False):
+  """ Returns a list of populations associated with the given dcids.
 
-  # Get the variable names
-  seed_col_var = '?' + seed_col_name.replace(' ', '_')
-  new_col_var = '?' + new_col_name.replace(' ', '_')
-  labels = {seed_col_var: seed_col_name, new_col_var: new_col_name}
+  The pv_map must be a dictionary mapping p (property) to v (value) in a pv
+  pair defining a population.
 
-  # Get the type of the container place
-  seed_col_type = self._col_types[seed_col_name]
-  new_col_type = 'StatisticalPopulation'
-  type_hint = {seed_col_var: seed_col_type, new_col_var: new_col_type}
+  If the dcids field is a list, then the return value is a dictionary mapping
+  dcid to the list of values associated with the given property.
 
-  # Get allowed DCIDs
-  dcids = list(self._dataframe[seed_col_name])
-  if not dcids:
-    # All entries in the seed column are empty strings. The new column should
-    # contain no entries.
-    self._dataframe[new_col_name] = ''
-    self._col_types[new_col_name] = new_col_type
-    return
-
-  # Construct the query
-  query = utils.DatalogQuery()
-  # Specify which variables to SELECT
-  query.add_variable(seed_col_var, new_col_var)
-  # Add constraints to the SELECT SQL query
-  query.add_constraint('?node', 'typeOf', seed_col_type)
-  query.add_constraint('?pop', 'typeOf', 'StatisticalPopulation')
-  query.add_constraint('?node', 'dcid', dcids)
-  query.add_constraint('?node', 'dcid', seed_col_var)
-  query.add_constraint('?pop', location_property, '?node')
-  query.add_constraint('?pop', 'dcid', new_col_var)
-  query.add_constraint('?pop', 'populationType', population_type)
-
-  pv_pairs = sorted(kwargs.items())
-  idx = 0
-  for idx, pv in enumerate(pv_pairs, 1):
-    query.add_constraint('?pop', 'p{}'.format(idx), pv[0]) # ? need str(pv[0])
-    query.add_constraint('?pop', 'v{}'.format(idx), pv[1]) # ditto
-  query.add_constraint('?pop', 'numConstraints', idx)
-
-  # Perform the query and merge the results
-  new_frame = Frame(datalog_query=query, labels=labels, type_hint=type_hint, rows=rows)
-  self.merge(new_frame)
-
-def get_observations(self,
-                     seed_col_name,
-                     new_col_name,
-                     observation_date,
-                     measured_property,
-                     stats_type=None,
```

Comment:
Nit: can you swap measured_property and obsevation_date, then move stats_type down? That way we have logical flow of observation_date followed by observation_period